pkg/scheduler/factory/factory.go

```diff
@@ -1085,6 +1085,7 @@ func MakeDefaultErrorFunc(client clientset.Interface, backoff *util.PodBackoff,
 		}
 
 		backoff.Gc()
+		cycle := podQueue.Cycle()
```

Comment:
I can see your point but adding an interface named "Cycle()" which will expose implementation detail of queue to outside doesn't look good.

May need to think twice.